### ISSUE TITLE
Question about running

### ISSUE THREAD

**Issue by yijuekip** [jonathanglevine/openfdashinyapps#2]
I have read with R-shiny app source code which you post on Github, But I cant run them on my Rstudio even I download the devtools. The error message is' ERROR: invalid multibyte character in parser at line 76'

Do you know How I can run them on my own R studio instead of the website?

**Comment by jonathanglevine** [comment]
It should work, since I developed it using RStudio.  Can you run other Shiny apps in RStudio?  Which version of R are you using?

**Comment by yijuekip** [comment]
the version is 3.3.1, and i try multiple APP to run 'server.R' file. Should I run other file?

**Comment by jonathanglevine** [comment]
That should work.  You do have the Shiny package installed, and you trying to run it via the “runApp” command, correct?

Also, can you figure out what is on line 76?

**Comment by yijuekip** [comment]
Yes , I can run sample code,  so my Shiny package should be good.  Different 'server.R' are different on line76. But i get same ERROR message. I have tried multiple 'server.R' in different documents.
what below is  message on console window 
> runApp('D:/PhD/summer intern 2018/Peter Assignment/openfdashinyapps-master/openfdashinyapps-master/foodrecallview')

Listening on http://127.0.0.1:5903
Warning in grepl("\n", lines, fixed = TRUE) :
  input string 76 is invalid in this locale
Warning in grepl("\n", lines, fixed = TRUE) :
  input string 169 is invalid in this locale
Warning in grepl("\n", lines, fixed = TRUE) :
  input string 265 is invalid in this locale
Warning in grepl("\n", lines, fixed = TRUE) :
  input string 275 is invalid in this locale
Warning: Error in source: invalid multibyte character in parser at line 76
Stack trace (innermost first):
    46: source
    45: eval [sourcedir.R#6]
    44: eval
    43: withVisible
    42: source
     1: runApp

**Comment by jonathanglevine** [comment]
Is the sharedscripts folder in the same folder as the foodrecallview folder?

**Comment by yijuekip** [comment]
yes, they are all in 'openfdashinyapps-master'  folder.

**Comment by jonathanglevine** [comment]
Do you have a key.r file in sharedscripts?  If not, put the attached script in sharedscripts.

**Comment by yijuekip** [comment]
there is no key.r file in sharescripts.  so where is the key.r file? I try to search key.r in 'openfdashinyapps-master' folder but no result come out...

**Comment by jonathanglevine** [comment]
It should have been attached to the e-mail I sent you.  It is a simple script:

getkey <- function() {
  return('')
}

**Comment by yijuekip** [comment]
I create 'key.R' in sharescripts folder with the script you gave me:
getkey <- function() {
  return('')
}
but the problem is not solved, same error message.

**Comment by jonathanglevine** [comment]
I think there are some characters in the help functions.r file that are invalid in your R installation.  Try this: in sourcedir.R comment out line 6, that is change

source( paste0( SOURCEDIR, 'helpfunctions.r') )

to 

#source( paste0( SOURCEDIR, 'helpfunctions.r') )

Hope this works.

**Comment by yijuekip** [comment]
I add'#' , now, the error message is 'ERROR: could not find function "bsButton"'

**Comment by jonathanglevine** [comment]
You need to install the shinyBS package.

**Comment by yijuekip** [comment]
and when i install bsbutton, it said bsbutton is not available for version 3.3.1

**Comment by jonathanglevine** [comment]
bsbutton is a function in the shinyBS package

**Comment by yijuekip** [comment]
i download shinyBS, the message is 
Warning: package ¡®shinyBS¡¯ was built under R version 3.3.3
Warning: Error in HTML: could not find function "loadhelp"

**Comment by jonathanglevine** [comment]
Not surprising.  The Shiny apps should still run with the warnings

**Comment by yijuekip** [comment]
I mean i click runAPP, error message is 'Error in HTML: could not find function "loadhelp"'

**Comment by yijuekip** [comment]
where can I find loadhelp? or do i need to change the version of R or R studio?

**Comment by jonathanglevine** [comment]
Okay, I think I know what is going on.  Try this:

1. Replace helpfunctions.r in sharedscripts.r  with the attached replacement help functions.r file
2. Change

#source( paste0( SOURCEDIR, 'helpfunctions.r') )

to 

source( paste0( SOURCEDIR, 'helpfunctions.r') )


Let know if there is a problem with the attached file.

If this doesn’t work, send me the text of the error message.

**Comment by yijuekip** [comment]
sorry, I am confused, where is 'attached replacement helpfunction.r'?

**Comment by jonathanglevine** [comment]
Is there a file called helpfunction.r attached to my last e-mail?

**Comment by yijuekip** [comment]
No, i didn't see the attachment with your e-mail.  both key.r and helpfunction.r......

**Comment by yijuekip** [comment]
just in case, you can send it to my email directly, yijue.wang@uconn.edu

**Comment by yijuekip** [comment]
Hi~I received your new helpfunction.r  and it works!!! thank you very much for your help!!!
so what is the problem? the character not fit to the version?
